packages/twilio-worker-auth/tests/unit/twilioWorkerAuthMiddleware.test.ts

```diff
@@ -1,12 +1,20 @@
+import { user } from '../../src';
+
 const tftv = require('twilio-flex-token-validator');
 import each from 'jest-each';
-import { getAuthorizationMiddleware } from '../../src/middlewares';
-import * as Permissions from '../../src/permissions';
+import { getAuthorizationMiddleware } from '../../src';
+import { unauthorized } from '@tech-matters/http';
+
+jest.mock('@tech-matters/http', () => ({
+  unauthorized: jest.fn(),
+}));
 
 beforeEach(() => {
   jest.clearAllMocks();
 });
 
+const mockUnauthorized = unauthorized as jest.Mock<ReturnType<typeof unauthorized>>;
```

Comment:
I have no idea what is going on here and how `mockUnauthorized` is being used. Could you shed some light?

Reply:
mockUnauthorized is just a casted version of unauthorized, which is already mocked in the jest.mock call above.

I've just found it useful to have a separate variable as a mock type for setting up the mocking, but keeping the original with its original typing for running expectations, cuts down on the need for inline casting in the tests

In this case we are just mocking an unauthorized response on line 82 and verifying the interaction occured as part of the test assertions

Reply:
Ahh! That make total sense thanks for clarifying.